test/RelayHubGas.test.ts

```diff
@@ -0,0 +1,1471 @@
+            factory = await createSmartWalletFactory(smartWalletTemplate);
+            recipientContract = await TestRecipient.new();
+            const testToken = artifacts.require('TestToken');
+            token = await testToken.new();
+            target = recipientContract.address;
+            verifier = verifierContract.address;
+            relayHub = relayHubInstance.address;
+
+            forwarderInstance = await createSmartWallet(
+                _,
+                gaslessAccount.address,
+                factory,
+                gaslessAccount.privateKey,
+                chainId
+            );
+            forwarder = forwarderInstance.address;
+            await token.mint('1000', forwarder);
+
+            sharedRelayRequestData = {
+                request: {
+                    relayHub: relayHub,
+                    to: target,
+                    data: '',
+                    from: gaslessAccount.address,
+                    nonce: (await forwarderInstance.nonce()).toString(),
+                    value: '0',
+                    gas: gasLimit,
+                    tokenContract: token.address,
+                    tokenAmount: '1',
+                    tokenGas: '50000'
+                },
+                relayData: {
+                    gasPrice,
+                    relayWorker,
+                    callForwarder: forwarder,
+                    callVerifier: verifier
+                }
+            };
+        });
+
+        context('with staked and registered relay', function () {
+            const url = 'http://relay.com';
+            const message = 'Enveloping RelayHub';
+            const messageWithNoParams = 'Method with no parameters';
+
+            let relayRequest: RelayRequest;
+            let encodedFunction: string;
+            let signatureWithPermissiveVerifier: string;
+
+            beforeEach(async function () {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('2'),
+                    from: relayOwner
+                });
+
+                // truffle-contract doesn't let us create method data from the class, we need an actual instance
+                encodedFunction = recipientContract.contract.methods
+                    .emitMessage(message)
+                    .encodeABI();
+
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.registerRelayServer(url, {
+                    from: relayManager
+                });
+                relayRequest = cloneRelayRequest(sharedRelayRequestData);
+                relayRequest.request.data = encodedFunction;
+
+                const dataToSign = new TypedRequestData(
+                    chainId,
+                    forwarder,
+                    relayRequest
+                );
+                signatureWithPermissiveVerifier = getLocalEip712Signature(
```

Comment:
signatureWithPermissiveVerifier -> variable is not being used

Reply:
fixed